project/src/main/java/com/google/sps/data/PlacesFetcher.java

```diff
@@ -108,6 +108,10 @@
                     .setRating(placeDetails.rating)
                     .setPriceLevel(Integer.parseInt(placeDetails.priceLevel.toString()))
                     .setLocation(placeDetails.geometry.location)
+                    .setPlaceId(placeDetails.placeId)
+                    .setGoogleUrl(placeDetails.url == null
+                            ? "" : placeDetails.url.toString())
```

Comment:
Guava has a `nullToEmpty` method which is very useful: https://guava.dev/releases/20.0/api/docs/com/google/common/base/Strings.html

There are many more useful methods there, so maybe you should add Guava to your project and use it for String manipulation.

Reply:
Great, I used this where possible. I also found a better solution for the URL to String conversion that wasn't elegant before because of the null case.